src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -94,73 +117,238 @@ boot_newcomp_sinv_alloc(spdid_t spdid)
 			user_cap_vaddr = (void *) (new_comp_cap_info[spdid].vaddr_mapped_in_booter + (new_comp_cap_info[spdid].vaddr_user_caps - new_comp_cap_info[spdid].addr_start) + (sizeof(struct usr_inv_cap) * i));
 
 			/* Create sinv capability from client to server */
-			sinv = cos_sinv_alloc(newcomp_compinfo, interface_compinfo->comp_cap, (vaddr_t)new_comp_cap_info[spdid].ST_user_caps[i].service_entry_inst, token);
+			sinv = cos_sinv_alloc(newcomp_compinfo, interface_compinfo->comp_cap,
+				(vaddr_t)new_comp_cap_info[spdid].ST_user_caps[i].service_entry_inst,
+				token);
+
 			assert(sinv > 0);
 
 			new_comp_cap_info[spdid].ST_user_caps[i].cap_no = sinv;
 
-			/* Now that we have the sinv allocated, we can copy in the symb user cap to correct index */
-			memcpy(user_cap_vaddr, &new_comp_cap_info[spdid].ST_user_caps[i], sizeof(struct usr_inv_cap));
+			/*
+			 * Now that we have the sinv allocated, we can copy in the symb user
+			 * cap to correct index
+			 */
+			memcpy(user_cap_vaddr, &new_comp_cap_info[spdid].ST_user_caps[i],
+					sizeof(struct usr_inv_cap));
 		}
 	}
 }
 
 static void
-boot_newcomp_create(spdid_t spdid, struct cos_compinfo *comp_info)
+boot_newcomp_definfo_init(spdid_t spdid, compcap_t cc, boot_comp_flag_t comp_flags)
+{
+
+	struct cos_defcompinfo *defci     = cos_defcompinfo_curr_get();
+	struct cos_aep_info    *sched_aep = cos_sched_aep_get(defci);
+	struct cos_aep_info    *child_aep = cos_sched_aep_get(new_comp_cap_info[spdid].defcompinfo);
+	struct cos_compinfo    *child_ci  = cos_compinfo_get(new_comp_cap_info[spdid].defcompinfo);
+
+	child_ci->comp_cap = cc;
+
+	if (comp_flags) {
+		child_aep->thd = cos_initthd_alloc(boot_info(), child_ci->comp_cap);
+		assert(child_aep->thd);
+
+		init_thds[spdid] = child_aep->thd;
+
+		child_aep->tc = cos_tcap_alloc(boot_info());
+		assert(child_aep->tc);
+
+		child_aep->rcv = cos_arcv_alloc(boot_info(), child_aep->thd, child_aep->tc,
+				boot_info()->comp_cap, sched_aep->rcv);
+		assert(child_aep->rcv);
+	} else {
+		child_aep->tc  = sched_aep->tc;
+		child_aep->rcv = sched_aep->rcv;
+	}
+
```

Comment:
If we are creating the root scheduler, `comp_flags & BOOT_COMP_FLAG_SCHED`, then we must provide that component with infinite tcap so we don't switch back to the llbooter component.